portfolio/src/main/java/com/google/sps/servlets/Greeting.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/greeting")
+public class Greeting extends HttpServlet {
+
+  private List<String> greetings;
+
+  public void init() {
+    greetings = new ArrayList<>();
+    greetings.add("Hi! Glad you're here. ");
+    greetings.add("Hi! Thanks for coming. ");
+    greetings.add("Welcome! ");
+    greetings.add("Salutations! ");
+    greetings.add("Howdy! ");
+    greetings.add("Hi! Nice of you to visit. ");
+  }
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String greeting = greetings.get((int) (Math.random() * greetings.size()));
+    response.setContentType("text/html;");
+    if(greetings.size() != 0) {
+      response.getWriter().println(greeting + "Here's a little about me...");
+    }
+    else {
+      response.getWriter().println("Hi! Here's a little about me...");
+    }
+  }
```

Comment:
This check should go before line 43, so you can avoid calling `greetings.get()` if it's empty. Also, it's a little more readable to use greetings.isEmpty() instead of greetings.size() != 0. Try something like:
```
String greeting = "Hi! ";  // use this as a default
if (greetings != null && !greetings.isEmpty()) {
    greeting = greetings.get(....);
}
response.setContentType(....);
// Now greeting is set either way, so we can keep this line as is
response.getWriter().println(greeting + "Here's a little about me...");
```
(the comments are just notes for you, you don't need to include them :) )
Does that make sense? So we're checking if it's empty before calling get(), because get() will throw an exception if the index is out of bounds. Also, I did see your comment that this will stay hard coded so this check isn't strictly necessary, it's just good practice :)

Reply:
yeah that makes sense